adler32.c

```diff
@@ -62,11 +61,24 @@ static uint32_t adler32_combine_(uint32_t adler1, uint32_t adler2, z_off64_t len
 #  define MOD63(a) a %= BASE
 #endif
 
+#if defined(X86_SSSE3_ADLER32)
+# include "arch/x86/x86.h"
+uint32_t ZLIB_INTERNAL adler32_ssse3(uint32_t adler, const unsigned char *buf, z_size_t len);
+#endif
+
 /* ========================================================================= */
-uint32_t adler32_c(uint32_t adler, const unsigned char *buf, size_t len) {
+uint32_t ZLIB_INTERNAL adler32_c(uint32_t adler, const unsigned char *buf, size_t len) {
     uint32_t sum2;
     unsigned n;
 
+#if defined(X86_SSSE3_ADLER32)
+    if (buf && len >= 64 && x86_cpu_has_ssse3)
+        return adler32_ssse3(adler, buf, len);
```

Comment:
Here is the change similar to the one under review in #303.

Reply:
The test for x86_cpu_has_ssse3 should be first so it does not even try other tests if that fails. It makes no sense for me to do two unnecessary tests if third fails.